features/setting_constraints/matching_arguments.feature

```diff
@@ -72,6 +72,36 @@ Feature: Matching arguments
       | expected: (a collection containing exactly 1 and 2)           |
       | got: ([1, 3])                                                 |
 
+  Scenario: Using satisfy for any expectation
+    Given a file named "rspec_satisfy_spec.rb" with:
+      """ruby
+      RSpec.describe "Using satisfy for any expectation" do
+        let(:dbl) { double }
+        before do
+          expectation = satisfy do |arg|
+            arg[:a] == 5
```

Comment:
Prehaps make this more complex to demonstrate why `satisfy` is needed

Reply:
This shows my own use case: asserting something about a value deep inside a blob of data.

Nested usage of `hash_including` would work for this use case, but becomes unwieldy fast with each level added.
 
```ruby
expectation = satisfy do |data|
  data.dig(:a, :b, :c) == 5
end
```

Reply:
Great, perfect.